src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -431,6 +432,9 @@ internal NamespaceOrTypeOrAliasSymbolWithAnnotations BindNamespaceOrTypeOrAliasS
                 case SyntaxKind.PointerType:
                     return bindPointer();
 
+                case SyntaxKind.FunctionPointerType:
+                    return TypeWithAnnotations.Create(FunctionPointerTypeSymbol.CreateFunctionPointerTypeSymbolFromSource((FunctionPointerTypeSyntax)syntax, this, diagnostics, basesBeingResolved));
```

Comment:
>TypeWithAnnotations.Create(FunctionPointerTypeSymbol.CreateFunctionPointerTypeSymbolFromSource((FunctionPointerTypeSyntax)syntax, this, diagnostics, basesBeingResolved)); [](start = 27, length = 170)

I think ```suppressUseSiteDiagnostics``` should be respected.

Reply:
I am not sure why we are being inconsistent about that in existing code as well. We either need to understand why it is Ok to be inconsistent or align all branches.

---
In reply to: [359051739](https://github.com/dotnet/roslyn/pull/40431#discussion_r359051739) [](ancestors = 359051739)

Reply:
> We either need to understand why it is Ok to be inconsistent or align all branches.

That's outside the scope of this PR, so I've filled an issue to track: [https://github.com/dotnet/roslyn/issues/40707](https://github.com/dotnet/roslyn/issues/40707)

---
In reply to: [359053343](https://github.com/dotnet/roslyn/pull/40431#discussion_r359053343) [](ancestors = 359053343,359051739)